samples/estimation/df-chemistry/chemistry.py

```diff
@@ -0,0 +1,482 @@
+import math
+import numpy as np
+import numpy.typing as npt
+import qsharp
+from argparse import ArgumentParser
+from dataclasses import dataclass
+from scipy import linalg as LA
+from urllib.parse import urlparse
+from urllib.request import urlretrieve
+
+
+@dataclass
+class FCIDumpFileContent:
+    num_orbitals: int
+    one_body_terms: list[tuple[list[int], float]]
+    two_body_terms: list[tuple[list[int], float]]
+
+    @classmethod
+    def from_file(self, file_url):
+        """Read FCIDump file (given by either a local path or a URL) and parse
+        it to get the input data for double-factorized chemistry algorithm."""
+        url = urlparse(file_url)
+
+        file_name = ''
+        if url.scheme in ['http', 'https']:
+            # Download the file
+            file_name = url.path.rsplit('/', 1)[-1]
+            print(f'Downloading {file_url} to {file_name}...')
+            urlretrieve(file_url, file_name)
+        elif url.scheme in ['', 'file']:
+            # Use the whole URL as the file path
+            file_name = file_url
+        else:
+            raise f"Unknown file location {file_url}"
```

Comment:
```suggestion
            raise f"Unsupported file location {file_url}"
```

The location might be known, but we do not support downloading it.